httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java

```diff
@@ -71,7 +71,7 @@ public IOSessionImpl(final String type, final SelectionKey key, final SocketChan
         this.commandQueue = new ConcurrentLinkedDeque<>();
         this.lock = new ReentrantLock();
         this.socketTimeout = Timeout.DISABLED;
-        this.id = String.format(type + "-%08X", COUNT.getAndIncrement());
+        this.id = String.format(type + "-%08d", COUNT.getAndIncrement());
```

Comment:
@michael-o `08d` would not be enough for a LONG value, would it? Should not it be 20?

Reply:
If the padding isn't important and we're okay with ambiguous substring matches, it could be worth considering simple concatenation to avoid poor String.format performance until we get something like [jep-348](https://openjdk.java.net/jeps/348).
```suggestion
        this.id = type + "-" + COUNT.getAndIncrement();
```

Reply:
@ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.

@carterkozak I can live with both. Whatever the community seems better readable/greppable.

Reply:
> @ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.

@michael-o Is not `9,223,372,036,854,775,807`longest possible LONG?

Reply:
> 
> 
> > @ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.
> 
> @michael-o Is not `9,223,372,036,854,775,807`longest possible LONG?

That's correct. I wanted to make a compromise. Having `ex-00000000000000000001` looks awkward.

Reply:
@michael-o Fair enough.

Reply:
Another option is to exclude padding in favor of a trailing separator for concise identifiers that cannot be ambiguous:

```java
this.id = type + "[" + COUNT.getAndIncrement() + "]";
```

This solves the problem grepping between `ex-1` and `ex-10` by producing values `ex[1]` and `ex[10]`, and in my opinion is easier for humans to quickly parse without N leading zero characters. What do you think?

Reply:
@carterkozak While you are right, it should read as an array. That's the whole point.